kvm/memory.go

```diff
@@ -0,0 +1,21 @@
+package kvm
+
+// UserSpaceMemoryRegion defines Memory Regions.
+type UserspaceMemoryRegion struct {
+	Slot          uint32
+	Flags         uint32
+	GuestPhysAddr uint64
+	MemorySize    uint64
+	UserspaceAddr uint64
+}
+
+// SetMemLogDirtyPages sets region flags to log dirty pages.
+// This is useful in many situations, including migration.
+func (r *UserspaceMemoryRegion) SetMemLogDirtyPages() {
+	r.Flags |= 1 << 0
+}
+
+// SetMemReadonly marks a region as read only.
+func (r *UserspaceMemoryRegion) SetMemReadonly() {
+	r.Flags |= 1 << 1
+}
```

Comment:
I think `SetUserMemoryRegion`, `SetTSSAddr` and `SetIdentityMapAddr` functions are needed here.

Reply:
done